### ISSUE TITLE
macOS target is too old?

### ISSUE THREAD

**Issue by Tieske** [luarocks/luarocks#1381]
### Please fill out the following:
- macOS Big Sur, 11.6.2
- luarocks: `/usr/local/Cellar/openresty@1.19.3.2/1.19.3.2/luarocks/bin/luarocks 3.5.0`

### Reproduce:
`luarocks install cqueues 20200726.51-0` fails with:

```
env MACOSX_DEPLOYMENT_TARGET=10.5 gcc -O2 -std=gnu99 -fPIC -g -Wall -Wextra -Wno-missing-field-initializers -Wno-initializer-overrides -Wno-unused -Wno-dollar-in-identifier-extension -Wno-deprecated-declarations -O2 -fPIC -D_REENTRANT -D_THREAD_SAFE -D_GNU_SOURCE -I"/usr/local/opt/openssl/include" -I"/usr/local/opt/openssl/include" -DSOCKET_DEBUG -DHAVE_CONFIG_H -DDNS_RANDOM=arc4random -c -o /private/var/folders/wv/1jbcxvf93pn1wk102hdpnzdm0000gn/T/luarocks_cqueues-20200726.51-0-4304143/cqueues-rel-20200726/src/lib/socket.o /private/var/folders/wv/1jbcxvf93pn1wk102hdpnzdm0000gn/T/luarocks_cqueues-20200726.51-0-4304143/cqueues-rel-20200726/src/lib/socket.c
/private/var/folders/wv/1jbcxvf93pn1wk102hdpnzdm0000gn/T/luarocks_cqueues-20200726.51-0-4304143/cqueues-rel-20200726/src/lib/socket.c:498:10: error: thread-local storage is not supported for the current target
                static __thread char sslstr[256];
                       ^
1 error generated.
make[1]: *** [/private/var/folders/wv/1jbcxvf93pn1wk102hdpnzdm0000gn/T/luarocks_cqueues-20200726.51-0-4304143/cqueues-rel-20200726/src/lib/socket.o] Error 1

Error: Failed installing dependency: https://luarocks.org/cqueues-20200726.51-0.src.rock - Build error: Failed building.
make: *** [dependencies] Error 1
$
```

### Workaround
```
export SYSTEM_VERSION_COMPAT=1
```

As suggested here: https://github.com/wahern/cqueues/issues/246#issuecomment-753579826 and related here: https://github.com/luarocks/luarocks/issues/1238

Seems the `MACOSX_DEPLOYMENT_TARGET=10.5` that LuaRocks injects is too old.

**Comment by daurnimator** [comment]
Any idea where it's getting 10.5 from? shouldn't it be 10.8 by default? (which is still very old)

**Comment by hishamhm** [comment]
There is [code in cfg.lua](https://github.com/luarocks/luarocks/blob/master/src/luarocks/core/cfg.lua#L442-L455) that tries to autodetect the macOS version it is running under, and sets it to `10.8` if it's running on 10.10 or higher. @Tieske, what does the command `sw_vers -productVersion`  output in your system?

As a workaround, you can do:

```bash
$ luarocks config variables.MACOSX_DEPLOYMENT_TARGET 10.8
```

to change your LuaRocks's installation default. It's also possible to set it in the rockspec like this:

```lua
build = {
   macosx_deployment_target = "10.10",
   -- ...
}
```

...if the rockspec uses features that are only available from a given macOS version onwards.

**Comment by Tieske** [comment]
@hishamhm here's the output from `sw_vers -productVersion`:
```
$ sw_vers -productVersion
11.6.3
$
```

**Comment by hishamhm** [comment]
Should we use "11.0" if running on 11.0 or above, I guess?

If you do 

```
$ luarocks config variables.MACOSX_DEPLOYMENT_TARGET 11.0
```

does that fix it?

**Comment by hishamhm** [comment]
Also, just noticed you were running LuaRocks 3.5.0. The latest LuaRocks should pick 10.8 by default, which might be sufficient?

**Comment by Tieske** [comment]
```
luarocks config variables.MACOSX_DEPLOYMENT_TARGET 11.0
```
did not fix the issue

**Comment by Tieske** [comment]
upgrading to 3.8.0 now builds the lib

**Comment by Tieske** [comment]
do get a bunch of warnings:
```
ld: warning: dylib (/usr/local/opt/openssl/lib/libssl.dylib) was built for newer macOS version (11.0) than being linked (10.8)
ld: warning: dylib (/usr/local/opt/openssl/lib/libcrypto.dylib) was built for newer macOS version (11.0) than being linked (10.8)
```

**Comment by hishamhm** [comment]
Thanks! Good to know it builds the library, but I guess we should we use still use "11.0" if running on 11.0 or above.

**Comment by hishamhm** [comment]
Closed via e06935214612d5c09dd4e0faf9bb28cb6ec6d895 !
